src/server.js

```diff
@@ -773,6 +777,10 @@ function performAuthentication(data) {
 	let client;
 	let token = null;
 
+	if (!Helper.config.public && Helper.config.headerAuth.enable) {
+		data.user = socket.handshake.headers[Helper.config.headerAuth.header];
```

Comment:
If the header isn't being set or the header name is incorrect, `data.user` will be set to `unknown` but The Lounge will get stuck because it can't work with that as a username. This can be tested/reproduced by setting `config.js` to look for a header name that doesn't exit.

Later on in the code, there is a check if `data.user` is not a string but I think that at this ^^ point, the code should behave as if the authorisation has failed, i.e. `socket.emit("auth:failed);`

Reply:
hopefully I've added enough checks